spec/tree-view-package-spec.coffee

```diff
@@ -3847,6 +3847,23 @@ describe "TreeView", ->
     files = Array.from(element.querySelectorAll('.entries .file'))
     files.find((file) -> file.fileName.textContent is text)
 
+describe "Service provider", ->
+  [provideTreeView, treeView] = []
+  beforeEach ->
+    waitForPackageActivation()
+
+    runs ->
+      provideTreeView = atom.packages.getActivePackage('tree-view').mainModule.provideTreeView()
+      treeView = atom.workspace.getLeftDock().getActivePaneItem()
+
+  it "should provide selected paths", ->
+    expect(provideTreeView.selectedPaths()).toEqual([atom.project.getPaths()[0]])
+
+  it "should provide entry for a path", ->
```

Comment:
The description of both of these tests should be a bit more meaningful I think, but sadly I am unable to churn out anything right now to help you. Maybe one of the other maintainers can chime in with some ideas on how to rewrite them.

/cc @atom/feedback

Reply:
I'm not sure what this should test other than that the method exists on the service. The internal methods should have their own tests to make sure they are actually working properly.